ctapipe/image/time_next_neighbor_cleaning.py

```diff
@@ -0,0 +1,742 @@
+
+    def fill_individual_pixel_rate(self, geo, image, trace_length, charge_steps):
+        """
+        Count the number of pixels in images about a given charge threshold
+        and fill dictionary with individual pixel noise rate (IPR).
+
+        Parameters
+        ----------
+        geo: ctapipe.instrument.CameraGeometry
+        image: numpy.ndarray
+        trace_length: astropy.units.Quantity
+            total length of the trace, e.g. number of samples times sample_time
+        charge_steps: numpy.ndarray
+            array with the charge steps used to calculate the IPR graphs.
+        """
+
+        self.IPR_dict = self.IPR_dict if self.IPR_dict else {}
+
+        if geo.camera_name not in self.IPR_dict:
+            self.IPR_dict[geo.camera_name] = {
+                "counter": 0,
+                "charge": charge_steps,
+                "npix": np.zeros(charge_steps.shape[0]),
+                "rate": np.zeros(charge_steps.shape[0]),
+                "rate_err": np.zeros(charge_steps.shape[0]),
+            }
+
+        # count number of pixels
+        self.IPR_dict[geo.camera_name]["counter"] += np.sum(image >= 0)
+        for thbin, charge in enumerate(self.IPR_dict[geo.camera_name]["charge"]):
+            self.IPR_dict[geo.camera_name]["npix"][thbin] += np.sum(image > charge)
+
+        # convert count of number of pixels above threshold into rate
+        sample_to_hz = (
+            1 / (trace_length * self.IPR_dict[geo.camera_name]["counter"])
+        ).to("Hz")
+        self.IPR_dict[geo.camera_name]["rate"] = (
+            self.IPR_dict[geo.camera_name]["npix"] * sample_to_hz
+        )
+        self.IPR_dict[geo.camera_name]["rate_err"] = (
+            np.sqrt(self.IPR_dict[geo.camera_name]["npix"]) * sample_to_hz
+        )
+
+    def calculate_ipr_from_source(self, source, calibrator, charge_steps):
+        """
+        Fill the IPR graph from a file or a list of files. The files should
+        contain pure night sky background calibration data. Therefore, the
+        calibration of the events in the source are performed. The calibrator
+        for this case should be equal to the one in the analysis.
+
+        Parameters
+        ----------
+        source:
+        calibrator:
+        charge_steps: numpy.ndarray
+            array storing the steps in charge used to create the IPR graph
+
+        """
+        for event in source:
+            calibrator.calibrate(event)
+            for tel_id in event.r0.tels_with_data:
+                geometry = event.inst.subarray.tel[tel_id].camera
+                image = event.dl1.tel[tel_id].image[0]
+
+                n_samples = event.r0.tel[tel_id].num_samples
+                sample_time = u.Quantity(event.mc.tel[tel_id].time_slice, u.ns)
+                sum_time = sample_time * n_samples
+
+                self.fill_individual_pixel_rate(geometry, image, sum_time, charge_steps)
+
+    def scaled_combfactor_event_display(self):
```

Comment:
This method is not used at all and contains values from EventDisplay "nobody undestands", so please just remove